src/iam.ts

```diff
@@ -1418,26 +1418,28 @@ export class IAM extends IAMBase {
     throw new CacheClientNotProvidedError();
   }
 
-  async subscribeToMessages({
-    topic = `${this._did}.${NATS_EXCHANGE_TOPIC}`,
+  async subscribeTo({
+    subject = `${this._did}.${NATS_EXCHANGE_TOPIC}`,
     messageHandler
   }: {
-    topic?: string;
+    subject?: string;
     messageHandler: (data: IMessage) => void;
   }) {
     if (!this._natsConnection) {
       return;
     }
-    this._exchangeSubscription = this._natsConnection.subscribe(topic);
-    for await (const msg of this._exchangeSubscription) {
+    const subscription = this._natsConnection.subscribe(subject);
+    this._subscriptions.push(subscription);
+    for await (const msg of subscription) {
       const decodedMessage = this._jsonCodec?.decode(msg.data) as IMessage;
       messageHandler(decodedMessage);
     }
   }
 
-  async unsubscribeFromMessages() {
-    if (this._exchangeSubscription) {
-      this._exchangeSubscription.unsubscribe();
+  async unsubscribeFrom(subject: string) {
```

Comment:
How can I unsubscribe to subscription 3 if I have below set of codes:
```
this.iam.subscribeTo({ messageHandler: this.myHandler1 }); // subscription 1
this.iam.subscribeTo({ messageHandler: this.myHandler2 }); // subscription 2
this.iam.subscribeTo({ messageHandler: this.myHandler3 }); // subscription 3
```

Reply:
Fixed. Subscriptions will be identified by id